CONTRIBUTING.md

```diff
@@ -1,10 +1,136 @@
-#Contributing
+# Contributing
+When contributing to this repository, please first discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change.
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+[Code of Conduct](#code-of-conduct) 
 
-For contributing please follow the instructions below:
-1. Fork the repository on GitHub
-2. Clone the project to your own machine
-3. Commit changes to your own branch
-4. push your work back up to your forked repo
-5. Submit a Pull Request so that we can review your changes
+# Pull Request Process
+1. Create Issue with relevant information about your PR
+2. Fork the repository on GitHub
+3. Clone the project to your own machine
+4. create a branch with relevent name for your pr
+5. Commit changes to your own branch
+6. push your work back up to your forked repo
+7. Submit a Pull Request so that we can review your changes
+8. Link the PR to the relevant issue
+9. Send the PR to the slack group and ask for reviewer
+10.  Ensure that your code follows the PEP 8 Style guide for Python code before submitting a pull request.
+    We have also set up Flake8 to ensure that our style guides are being followed(PR's that dont pass this may not get reviewed until those are solved)
 
-NOTE: Be sure to fetch and merge the latest changes from the "upstream" repository before making a pull request!
+#### Pull request template:
+1. Making changes/Creating XFILE
+2. Empty line
+3. What?
+4. Why?
+5. How?
+6. Testing?
+7. Screenshots(if needed)
+8. Anything else?
```

Comment:
Do you intend to keep this line? wont it be confusing to contributor?

more generically, PRs are not kept in git history (besides their merge commit)
so I usually advocate for good, atomic commits, with good commit msgs
then the PR itself can be simple in most cases.